pkg/providers/tinkerbell/comparator_test.go

```diff
@@ -0,0 +1,98 @@
+package tinkerbell_test
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+
+	"github.com/aws/eks-anywhere/pkg/providers/tinkerbell"
+	"github.com/aws/eks-anywhere/pkg/utils/ptr"
+)
+
+func TestKubeadmControlPlaneKubernetesVersionUnchanged(t *testing.T) {
+	g := NewWithT(t)
+
+	oldKCPComparator := &tinkerbell.KubeadmControlPlaneComparator{comparatorKubeadmControlPlane()}
+	newKCPComparator := &tinkerbell.KubeadmControlPlaneComparator{comparatorKubeadmControlPlane()}
+
+	g.Expect(tinkerbell.KubernetesVersionChange(oldKCPComparator, newKCPComparator), false)
+}
+
+func TestKubeadmControlPlaneKubernetesVersionChanged(t *testing.T) {
+	g := NewWithT(t)
+
+	oldKubeadmControlPlane := comparatorKubeadmControlPlane()
+	newKubeadmControlPlane := comparatorKubeadmControlPlane()
+	newKubeadmControlPlane.Spec.Version = "1.23"
+	oldKCP := &tinkerbell.KubeadmControlPlaneComparator{oldKubeadmControlPlane}
+	newKCP := &tinkerbell.KubeadmControlPlaneComparator{newKubeadmControlPlane}
+
+	g.Expect(tinkerbell.KubernetesVersionChange(oldKCP, newKCP), true)
+}
+
+func TestKubeadmControlPlaneReplicasUnchanged(t *testing.T) {
+	g := NewWithT(t)
+
+	oldKCPComparator := &tinkerbell.KubeadmControlPlaneComparator{comparatorKubeadmControlPlane()}
+	newKCPComparator := &tinkerbell.KubeadmControlPlaneComparator{comparatorKubeadmControlPlane()}
+
+	g.Expect(tinkerbell.ReplicasChange(oldKCPComparator, newKCPComparator), false)
```

Comment:
Should a `boolean` be expected here?

Reply:
Ah, that should definitely be a int. Thank you.